### ISSUE TITLE
When logging in/provisioning user, require at least name, email, uid

### ISSUE THREAD

**Issue by baszoetekouw** [SURFscz/SBS#773]
In some cases, users with an incomplete attributes set can log in or become a member of a CO. Usually this is caused by the initial login flow being interrupted at some point (usually during 2FA) and then the accepting of the invite being restarted without triggering a full login flow.  This has also occurred as a result of OP upstream disruptions and/or bugs.

So, please make the `name`, `email` and `uid` required for all SBS operations, specifically:
 - logging in to SBS
 - accepting a invitation (for CO, Org, Service)
 - logging in to a Service (i.e., requests to the `proxy_authz` endpoint)

In case the user is taking an action in SBS itself, clear the user's session, log them out, and show an error page:

> Your user record seems to be incomplete.  
> Please try logging in again (possibly is a fresh or incognito browser session).  
> If the problem reoccurs, please contact support@sram.surf.nl

If the problem occurs when the user is is trying to log in to a different service (via a `proxy_authz` call), send an interrupt response and redirect the user to the same error page.

In all cases, treat this as a critical issue in the logs, and make sure a log mail is send to sram-beheer.

**Comment by baszoetekouw** [comment]
See also #743

**Comment by oharsta** [comment]
@baszoetekouw  Will we receive the `name` attribute in the `proxy_authz` endpoint? This is where provision users if the user is unknown.

**Comment by oharsta** [comment]
The page where incomplete users are redirected to is /missing-attributes

**Comment by oharsta** [comment]
For now the assumption is made that the name of the user is provided in the `proxy_authz` as attribute `user_name`. See https://github.com/SURFscz/SBS/commit/03361e5a3341a51490456e3f4c8482861a15f011#diff-7425bc5eed5e90ab656daf788f5f43bec744d5c0a62adbd4397274abf05d66b3R240

**Comment by baszoetekouw** [comment]
I'm don't think that we should get a username from eduteams in the proxy_authz call.  In almost all flows a user first logs in to SBS (and thus gets a full OIDC user_info record and is provisioned correctly) before they can do anything.

I think we should refactor the login process altogether, and also get rid of a lot of the second factor workarounds that we have introduced in the past years.  To be continues in #883 

In any case, I've just tested the flow for a fresh user without a displayname in the proxy_auth call, and that works fine in the current implementation.
